src/sonic-yang-models/yang-models/sonic-sflow.yang

```diff
@@ -0,0 +1,109 @@
+module sonic-sflow{
+
+    namespace "http://github.com/Azure/sonic-sflow";
+    prefix sflow;
+    yang-version 1.1;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-types {
+        prefix stypes;
+    }
+    import sonic-port {
+        prefix prt;
+    }
+    import sonic-vlan {
+        prefix svlan;
+    }
+
+    description "SFLOW yang Module for SONiC OS";
+
+    revision 2021-04-26 {
+        description "First Revision";
+    }
+
+    container sonic-sflow {
+
+        container SFLOW_COLLECTOR {
+            list SFLOW_COLLECTOR_LIST {
+                max-elements 2;
+                key "name";
+
+                leaf name {
+                    type string {
+                        length 1..64;
+                    }
+                }
+
+                leaf collector_ip {
+                    mandatory true;
+                    type inet:ip-address;
+                }
+
+
+                leaf collector_port {
+                    type inet:port-number;
+                }
+
+            } /* end of list SFLOW_COLLECTOR_LIST */
+        } /* end of container SFLOW_COLLECTOR */
+
+        container SFLOW_SESSION {
+            list SFLOW_SESSION_LIST {
+                key "port";
+
+                leaf port {
+                    type leafref {
+                        path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:name";
+                    }
+                }
+
+                leaf admin_state {
+                    type stypes:admin_status;
+                }
+
+                leaf sample_rate {
+                    type uint32 {
+                        range "256..8388608" {
+                            error-message "sFlow sample rate must be [256-8388608]";
+                        }
+                    }
+                }
+            } /* end of list SFLOW_SESSION_LIST */
+        } /* end of container SFLOW_SESSION */
+
+        container SFLOW {
+
+            container global {
+
+                leaf admin_state {
+                    type stypes:admin_status;
+                }
+
+                leaf polling_interval {
+                    type uint16 {
+                        range "0|5..300" {
+                            error-message "sFlow polling interval must be [0, 5-300]";
+                        }
+                    }
+                }
+
+                leaf agent_id {
+                    type union {
+                        type leafref {
+                            path "/prt:sonic-port/prt:PORT/prt:PORT_LIST/prt:name";
+                        }
+                        type leafref {
+                            path "/svlan:sonic-vlan/svlan:VLAN/svlan:VLAN_LIST/svlan:name";
+                        }
```

Comment:
Please plan to add mgmt & loopback interfaces as well whenever those YANG models are available.

Reply:
I've added mgmt interface type and have commented it for now. Will enable it once the pending PR is merged, https://github.com/Azure/sonic-buildimage/pull/7252.

Regarding loopback interface, when the support for LOOPBACK table is in community, will add it here.